dtfs-central-api/src/helpers/get-keying-sheet-calculation-tfm-facility-values.test.ts

```diff
@@ -0,0 +1,173 @@
+        new NotFoundError(`TFM facility with ukefFacilityId '${facilityId}' could not be found`),
+      );
+    });
+  });
+
+  describe('when the found tfm facility does not contain a cover start date', () => {
+    beforeEach(() => {
+      findOneByUkefFacilityIdSpy.mockResolvedValue({
+        ...aTfmFacility(),
+        facilitySnapshot: {
+          ...aFacility(),
+          ukefFacilityId: facilityId,
+          interestPercentage: 5,
+          dayCountBasis: 365,
+          coverEndDate: today,
+          coverStartDate: null,
+        },
+      });
+    });
+
+    it('should throw a NotFoundError', async () => {
+      // Act / Assert
+      await expect(getKeyingSheetCalculationTfmFacilityValues(facilityId, reportPeriod)).rejects.toThrow(
+        new NotFoundError(`Failed to find a cover start date for the tfm facility with ukef facility id '${facilityId}`),
+      );
+    });
+  });
+
+  describe('when the found tfm facility does not contain a cover end date', () => {
+    beforeEach(() => {
+      findOneByUkefFacilityIdSpy.mockResolvedValue({
+        ...aTfmFacility(),
+        facilitySnapshot: {
+          ...aFacility(),
+          ukefFacilityId: facilityId,
+          interestPercentage: 5,
+          dayCountBasis: 365,
+          coverEndDate: null,
+          coverStartDate: today,
+        },
+      });
+    });
+
+    it('should throw a NotFoundError', async () => {
+      // Act / Assert
+      await expect(getKeyingSheetCalculationTfmFacilityValues(facilityId, reportPeriod)).rejects.toThrow(
+        new NotFoundError(`Failed to find a cover end date for the tfm facility with ukef facility id '${facilityId}`),
+      );
+    });
+  });
+
+  describe('when the report period is provided', () => {
+    beforeEach(() => {
+      findOneByUkefFacilityIdSpy.mockResolvedValue(mockFacility);
+    });
+
+    describe('when the facility does not contain a completed amendment', () => {
+      it("should return a populated object with the facility's cover end date", async () => {
+        const result = await getKeyingSheetCalculationTfmFacilityValues(facilityId, reportPeriod);
+
+        const expected = {
+          coverEndDate: convertTimestampToDate(today),
+          coverStartDate: convertTimestampToDate(today),
+          dayCountBasis: 365,
+          interestPercentage: 5,
+          coverPercentage: 5,
+          value: mockFacility.facilitySnapshot.value,
+        };
+
+        expect(result).toEqual(expected);
+      });
+    });
+
+    describe('when the facility contains a completed amendment', () => {
+      const dateAfterReportPeriodEnd = new Date(addDays(today, 1));
+
+      const mockFacilityWithAmendment = {
+        ...mockFacility,
+        amendments: [
+          {
+            ...aCompletedTfmFacilityAmendment(),
+            value: 350000,
+            effectiveDate: dateAfterReportPeriodEnd.getTime(),
```

Comment:
We should only be returning values from amendments which are effective before the end of the report period. Why is this test passing?

Reply:
I am not sure how this works.  I did not add this bit of code to this function for amendments and am just testing it since there was missing test coverage on the old function

Reply:
The test coverage was in `getFixedFeeForFacility` tests, since those no longer exists it needs to have the coverage that was in there via the parent function

Reply:
(See lines 184 to line 253 of the get-fixed-fee-for-facility.test.ts file you deleted - it makes use of mocking so doesn't need to actually maintain a true values for the amendment being passed in but since you don't like mocking you will need to find a way to satisfy that function)

Reply:
FWIW I think here it would just be fine to rename the variable to `dateBeforeReportPeriodEnd`

Reply:
changed to be one day before